### PR TITLE
feat: add eof support

### PR DESCRIPTION
**Pull request by barnabasbusa**

### CODE REVIEW

File: src/package_io/input_parser.star [hunk 1]
```diff
@@ -786,6 +787,7 @@ def default_network_params():
         "electra_fork_epoch": 100000000,
         "eip7594_fork_epoch": 100000001,
         "eip7594_fork_version": "0x70000038",
+        "eof_activation_epoch": "",
```
Comment (parithosh) [inline src/package_io/input_parser.star:790]:
```suggestion
        "eof_activation_epoch": "100000002",
```

File: src/prelaunch_data_generator/el_cl_genesis/el_cl_genesis_generator.star [hunk 2]
```diff
@@ -46,29 +26,9 @@ def generate_el_cl_genesis_data(
...
         genesis_unix_timestamp,
-        network_id,
-        deposit_contract_address,
-        seconds_per_slot,
-        preregistered_validator_keys_mnemonic,
         total_num_validator_keys_to_preregister,
-        genesis_delay,
-        max_per_epoch_activation_churn_limit,
-        churn_limit_quotient,
-        ejection_balance,
```
Comment (parithosh) [inline src/prelaunch_data_generator/el_cl_genesis/el_cl_genesis_generator.star:57]: removing all of this?

File: src/package_io/input_parser.star (hunk 1 above)
Comment (barnabasbusa) [reply src/package_io/input_parser.star:790]: No, as that can break ELs that don't have the praugeEOFTime defined yet.

File: src/prelaunch_data_generator/el_cl_genesis/el_cl_genesis_generator.star (hunk 2 above)
Comment (barnabasbusa) [reply src/prelaunch_data_generator/el_cl_genesis/el_cl_genesis_generator.star:57]: little cleanup. was a messy set, now I just pass network_params.

File: src/package_io/input_parser.star (hunk 1 above)
Comment (parithosh) [reply src/package_io/input_parser.star:790]: ugh

**Review by parithosh — APPROVED** [review]

**Pull request merged**